### ISSUE TITLE
Model trained on cityscapes projects static objects to infinity

### ISSUE THREAD

**Issue by aj96** [nianticlabs/monodepth2#40]
Hello,

I added support for training monodepth2 on cityscapes, and trained it using the default hyperparameters used to train for monocular on kitti. It looks like it is having trouble masking out the all the static pixels and is projecting cars that move with the ego-car to infinity. I was wondering if anyone had any useful input on this? The most obvious thing to do is to weight the static mask in such a way that it masks out more pixels as static. But I was wondering if the authors ran into this issue when training on kitti. It's possible that this didn't happen on kitti because the environment is mostly static? 

<img width="266" alt="initial_cityscapes_result" src="https://user-images.githubusercontent.com/19962279/60927884-45049880-a260-11e9-9c8e-a2deb64507b1.png">

**Comment by mdfirman** [comment]
Hi @aj96 ,

Thanks for trying this. Cityscapes is in general a much harder dataset to learn from, with more moving objects, so you shouldn't expect results to be as good as for KITTI.  This is true even for stereo training, where moving objects aren't an issue – compare for example the `CS` row (5th from top) with the `K` row (bottom) from this table from monodepth 1:

![image](https://user-images.githubusercontent.com/6114258/60963253-32459e80-a308-11e9-97b5-999f8ad3739d.png)

While I don't have any real tips to make great depths *everywhere*, the good news is: There is still more work to do in monocular depth estimation! If you manage to fix this problem on cityscapes I am sure the wider community would like to hear about it.

Thanks

**Comment by mdfirman** [comment]
I'm going to re-open this for a week or so in case anyone else visiting has any suggestions

**Comment by aj96** [comment]
@mdfirman How did you evaluate the accuracy on cityscapes? It doesn't appear that they provide lidar for ground truth like they do for kitti.

**Comment by mrharicot** [comment]
@aj96 They have precomputed disparities using SGM. While we can't really consider this to be a ground truth it's a good signal to monitor while training.

**Comment by mdfirman** [comment]
The results table though is evaluated on the KITTI lidar. So the published results are also measuring the ability of the CS model to transfer across datasets, which isn't ideal, but gives a reasonable idea of if it has learned something reasonable or not!

**Comment by ezorfa** [comment]
Hi @aj96 !

Could you please share the code for Cityscapes dataset to feed into data loader?
